projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1475,6 +1476,38 @@ public function publicize_checkbox_default( $checked, $post_id ) {
 		return $checked;
 	}
 
+	/**
+	 * Checks whether sig is enabled for a post.
+	 *
+	 * @param int $post_id ID of the current post.
+	 * @return bool True if sig is enabled for the post.
+	 */
+	private function is_sig_enabled_for_post( $post_id ) {
+		$social_options = get_post_meta( $post_id, '_wpas_options', true );
+		return $social_options['image_generator_settings']['enabled'];
```

Comment:
I've [added a method for checking the status to the SIG library](https://github.com/Automattic/jetpack/pull/29118) so we'll be able to use that instead.

As an aside, it's always good to check that a variable exists and has the same type as you're expecting before using it. In this case, if `$social_options` wasn't an array or didn't have the `image_generator_settings` key, this would throw a warning.